code-challenges.js

```diff
@@ -9,60 +9,151 @@
 
-const myNumbers = [13, 34, 42, 5, 5, 10, 27, 42, 10]
-const givenValue1 = 42
-// Expected output: 7
+//--------------------------
 
-const givenValue2 = 10
-// Expected output: 8
+// const myNumbers = [13, 34, 42, 5, 5, 10, 27, 42, 10]
 
-// --------------------5) Create the code that will sort all the numbers in an array from largest to smallest. Use the test variables provided below.
+// const givenValue1 = 42
+// // // Expected output: 7
+// // console.log(myNumbers.lastIndexOf(givenValue1))
+
+// const givenValue2 = 10
+// // // Expected output: 8
+// // console.log(myNumbers.lastIndexOf(givenValue2))
+// // -------------------------------
+// const test = "Test"
+
+// const givenValueArray = [givenValue1, givenValue2, test]
+
+// // This was my first attempt,  I miss understood the question when i read the word return.  My mind went to creating a function.  In this approach my if statement returned the correct value, but my else if statement did not return a value.  It looks like after the loop evaluated myNumber[2] it preformed the  the expected behavior. but did not evaluate the remaindng indexes in the array.  
+
+// const lastIndex = (array1) => {
+//     for (let i = 0; i < myNumbers.length; i++) {
+//     if (myNumbers[i] === array1) {
+// return myNumbers.lastIndexOf(givenValue1);
+
+// } else if (myNumbers[i] === array1) {
+// return myNumbers.lastIndexOf(givenValue2);
+// } 
+// }
+// }
+
+// console.log(lastIndex(givenValueArray[0]));
+// console.log(lastIndex(givenValueArray[1]));
+// console.log(lastIndex(givenValueArray[2]));
+
+//--------------------------
+
+// // --------------------5) Create the code that will sort all the numbers in an array from largest to smallest. Use the test variables provided below.
+
+// // Pseudo code: 1.) use a built in method to mutate arrays from largest to smallest.
+//                 2.) use .sort() method to sort the array from smallest to largetst
+//                 3.) use .reverse() method to flip the order of the array.
+
+// const sanDiegoSummerTemperatures = [79, 80, 72, 73, 82, 77, 76]
+// // // Expected output: [82, 80, 79, 77, 76, 73, 72]
+
+
+// const sanDiegoWinterTemperatures = [59, 68, 62, 59, 66, 67, 66]
+// // // Expected output: [68, 67, 66, 66, 62, 59, 59]
+
+
+// const dog = "test"
+
+// const combineArray = [sanDiegoSummerTemperatures, sanDiegoWinterTemperatures, dog]
 
-// Pseudo code:
 
-const sanDiegoSummerTemperatures = [79, 80, 72, 73, 82, 77, 76]
-// Expected output: [82, 80, 79, 77, 76, 73, 72]
+// const newArray = (array1) => {
+//     for (let i = 0; i < combineArray.length; i++){
+//     if (array1 === sanDiegoSummerTemperatures) {
+//         return sanDiegoSummerTemperatures.sort().reverse()
+//     } else if (array1 === sanDiegoWinterTemperatures) {
+//         return sanDiegoWinterTemperatures.sort().reverse()
+//     } else {
+//         return "It's raining cats and dogs!"
+//     }
+// }
```

Comment:
I love the fact that you are thinking of using a function for this problem! One of the things to consider is that when we see something that seems to be repeating, we may be able to refactor and make our code more dynamic. Great job using sort and reverse.